src/backend/distributed/connection/remote_commands.c

```diff
@@ -372,6 +372,20 @@ ExecuteCriticalRemoteCommandList(MultiConnection *connection, List *commandList)
  */
 void
 ExecuteCriticalRemoteCommand(MultiConnection *connection, const char *command)
+{
+	ExecuteCriticalRemoteCommandWithResult(connection, command, NULL);
+}
+
+
+/*
+ * ExecuteCriticalRemoteCommandWithResult executes a remote command that is
+ * critical to the transaction. If the command fails then the transaction
+ * aborts. This function also allows you to get the result of the query back.
+ * If NULL is passed as the result, the result will be automatically cleared.
+ */
+void
+ExecuteCriticalRemoteCommandWithResult(MultiConnection *connection, const char *command,
```

Comment:
This API seems a little unexpected. `GetRemoteCommandResult` may need be called multiple times to fetch the results. Here you can safely call for your purpose because you expect a single tuple result. 

So, I'd rather have a specific function for that instead of having a generic `ExecuteCriticalRemoteCommandWithResult` such as `TotalRelationSizeForSubscription`

Reply:
These changes are gone after removing the "retry loop"